File: android/app/src/main/java/com/galaxyrn/logger/GxyLoggerUtils.java
package com.galaxyrn.logger;

import android.os.Build;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/**
 * Utility class for GxyLogger containing message formatting and helper
 * functionality
 */
public class GxyLoggerUtils {

    private static final String TAG = "GxyLoggerUtils";

    // Date formatter for timestamps
    private static final SimpleDateFormat dateFormat = new SimpleDateFormat("HH:mm:ss.SSS", Locale.getDefault());
    // ========================================
    // PUBLIC UTILITY METHODS
    // ========================================

    /**
     * Format log message with additional info (always includes timestamp, thread
     * info, and method info)
     */
    public static String formatMessage(String message) {
        StringBuilder formatted = new StringBuilder();

        // Add timestamp (always)
        formatted.append("[").append(dateFormat.format(new Date())).append("] ");

        // Add thread info (always)
        Thread currentThread = Thread.currentThread();
        formatted.append("[Thread:").append(currentThread.getName()).append("] ");

        // Add method info (always)
        StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
        if (stackTrace.length > 6) {
            StackTraceElement caller = stackTrace[6]; // Skip this class and logger methods
            String className = caller.getClassName();
            String simpleClassName = className.substring(className.lastIndexOf('.') + 1);
            formatted.append("[")
                    .append(simpleClassName)
                    .append(".")
                    .append(caller.getMethodName())
                    .append(":")
                    .append(caller.getLineNumber())
                    .append("] ");
        }

        formatted.append(message);
        return formatted.toString();
    }

    /**
     * Log device information (useful for debugging)
     */
    public static void logDeviceInfo(String defaultTag) {
        GxyLogger.i(defaultTag, "=== DEVICE INFORMATION ===");
        GxyLogger.i(defaultTag, "Manufacturer: " + Build.MANUFACTURER);
        GxyLogger.i(defaultTag, "Model: " + Build.MODEL);
        GxyLogger.i(defaultTag, "Android Version: " + Build.VERSION.RELEASE);
        GxyLogger.i(defaultTag, "API Level: " + Build.VERSION.SDK_INT);
        GxyLogger.i(defaultTag, "========================");
    }
}

File: ios/GalaxyRN/Logger/GxyLoggerUtils.swift
import Foundation
import UIKit
import os.log

/**
 * Utility class for GxyLogger containing Sentry integration,
 * message formatting, and other helper functionality
 */
class GxyLoggerUtils {
    
    private static let TAG = "GxyLoggerUtils"
    
    // Date formatters for timestamps
    private static let dateFormatter: DateFormatter = {
        let formatter = DateFormatter()
        formatter.dateFormat = "HH:mm:ss.SSS"
        formatter.locale = Locale.current
        return formatter
    }()
    
    // MARK: - PUBLIC UTILITY METHODS
    
    /**
     * Format log message with additional info (always includes timestamp, thread
     * info, and method info)
     */
    static func formatMessage(_ message: String) -> String {
        var formatted = ""
        
        // Add timestamp (always)
        formatted += "[\(dateFormatter.string(from: Date()))] "
        
        // Add thread info (always)
        let currentThread = Thread.current
        let threadName = currentThread.isMainThread ? "Main" : (currentThread.name ?? "Background")
        formatted += "[Thread:\(threadName)] "
        
        // Add method info (always)
        let stackTrace = Thread.callStackSymbols
        if stackTrace.count > 6 {
            let caller = stackTrace[6]
            // Parse the stack trace to get class and method info
            if let range = caller.range(of: " ") {
                let methodInfo = String(caller[range.upperBound...])
                if let spaceRange = methodInfo.range(of: " ") {
                    let className = String(methodInfo[..<spaceRange.lowerBound])
                    formatted += "[\(className)] "
                }
            }
        }
        
        formatted += message
        return formatted
    }
    
    /**
     * Log device information (useful for debugging)
     */
    static func logDeviceInfo(_ defaultTag: String = TAG) {
        GxyLogger.i(defaultTag, "=== DEVICE INFORMATION ===")
        GxyLogger.i(defaultTag, "Device Model: \(UIDevice.current.model)")
        GxyLogger.i(defaultTag, "Device Name: \(UIDevice.current.name)")
        GxyLogger.i(defaultTag, "System Name: \(UIDevice.current.systemName)")
        GxyLogger.i(defaultTag, "System Version: \(UIDevice.current.systemVersion)")
        
        if let infoDictionary = Bundle.main.infoDictionary {
            if let appVersion = infoDictionary["CFBundleShortVersionString"] as? String {
                GxyLogger.i(defaultTag, "App Version: \(appVersion)")
            }
            if let buildNumber = infoDictionary["CFBundleVersion"] as? String {
                GxyLogger.i(defaultTag, "Build Number: \(buildNumber)")
            }
        }
        
        GxyLogger.i(defaultTag, "========================")
    }
}
